File: app/src/main/java/com/bracketcove/postrainer/reminderlist/ReminderListContract.java
package com.bracketcove.postrainer.reminderlist;

import com.bracketcove.postrainer.base.BasePresenter;
import com.bracketcove.postrainer.base.BaseView;
import com.bracketcove.postrainer.data.viewmodel.Reminder;

import java.util.List;

/**
 * Created by Ryan on 06/03/2017.
 */

public interface ReminderListContract  {
    interface View extends BaseView<Presenter> {
        void setReminderListData(List<Reminder> reminderListData);

        void setNoReminderListDataFound();

        void addNewReminderToListView(Reminder reminder);

        void undoDeleteReminderAt(int index, Reminder reminder);

        void startReminderDetailActivity(String reminderId);

        void startSettingsActivity();


    }

    interface Presenter extends BasePresenter {
        void onReminderToggled(boolean active, Reminder reminder);

        void onSettingsIconClick();

        void onReminderSwiped(int index, Reminder reminder);

        void onReminderIconClick(Reminder reminder);

        void onCreateReminderButtonClick(int currentNumberOfReminders, Reminder reminder);
    }
}

File: app/src/main/java/com/bracketcove/postrainer/reminderlist/ReminderListContract.kt
package com.bracketcove.postrainer.reminderlist

import com.wiseassblog.domain.domainmodel.Reminder

/**
 * Created by Ryan on 06/03/2017.
 */

interface ReminderListContract {
    interface View {
        fun setReminderListData(reminderListData: List<Reminder>)

        fun setNoReminderListDataFound()

        fun startReminderDetailView(reminderId: String)

        fun startSettingsActivity()

        fun showMessage(msg: String)

        fun startMovementsView()
    }
}

sealed class ReminderListEvent {
    object OnSettingsClick : ReminderListEvent()
    object OnMovementsClick : ReminderListEvent()
    object OnCreateButtonClick : ReminderListEvent()
    data class OnReminderIconClick(val reminder: Reminder) : ReminderListEvent()
    data class OnReminderToggled(val isActive: Boolean, val reminder: Reminder) : ReminderListEvent()
    object OnStart : ReminderListEvent()
}